components/sftp/sftp.app.js

```diff
@@ -0,0 +1,70 @@
+module.exports = {
+  type: "app",
+  app: "sftp",
+  props: {
+    db: "$.service.db"
+  },
+  methods: {
+    config: {
+      host: null,
+      port: null,
+      username: null,
+      password: null,
+    },
+    async getClient(deps) {
+      const ssh2SftpClient = require('ssh2-sftp-client'); // src: https://github.com/theophilusx/ssh2-sftp-client/blob/master/src/index.js
+      const sftp = new ssh2SftpClient();
+
+      // dependency injection for integration tests
+      if (deps) {
```

Comment:
I think you could cut down on the conditional logic here and also allow users to pass a partial config by doing something like:

```
await sftp.connect({
  host: deps.host || this.$auth.host,
  port: deps.port || this.$auth.port,
  ...
})
```

That way I can call this method with a config like `{ host: "192.186.1.1" }` to just set the host, falling back to `this.$auth`.

`this.$auth` and its properties should always exist because Pipedream requires users link an account before component deploy, and also requires that users enter all required params for a linked account, so you should have the params you need in `this.$auth` as fallbacks.

Not wed to this idea, just thinking about ways to cut down on code.